### DISCUSSION TITLE
Using "dataExplorer.scriptAsAlter" command

### DISCUSSION THREAD

**Discussion by MikhailProfile** [microsoft/azuredatastudio#23946] [Q&A] [answered]
Good day,

I'm trying to use  `vscode.commands.executeCommand("dataExplorer.scriptAsAlter",...)` in my custom extension. But I can't find any info how to run it properly. I tried to pass `azdata.ObjectExplorerContext `with no luck. 

Can somebody help with this command? How to use it properly?

**Comment by Charles-Gagnon** [comment, marked as answer]
What exactly are you trying to do? That's an internal command so isn't recommended to use itself - it will likely be difficult to get the correct context and is also something that can change at any point since we don't promise a contract around internal commands.

Here's the code for the command if you're interested : 
https://github.com/microsoft/azuredatastudio/blob/08311728b9c240f1b3f290b5821e9ffd36301f7c/src/sql/workbench/contrib/scripting/browser/scriptingActions.ts#L159

But you should be able to get the ScriptingProvider directly by calling `azdata.dataprotocol.getProvider<azdata.ScriptingProvider>('MSSQL', azdata.DataProviderType.ScriptingProvider);` and then calling `scriptAsOperation` on it (you'll need to fill in the info about the object being scripted, which if you have the node context should be contained in that)

Note - be careful not to use the `registerOnScriptingComplete` function or you'll break the core ADS functionality due to https://github.com/microsoft/sqlops-dataprotocolclient/issues/62

**Reply by MikhailProfile** [reply]
Good day Charles, 

Thanks for your reply!

My idea is to have a QuickPick where I can search for all objects in my MSSQL including source codes of SP,Functions etc(Done). On pressing enter I want to open in a new editor "ALTER script" if it's SP or Function or "Select top 1000" if it's a Table or View. 

For now I can open editor and put code of procedure there but it's a "create" code.  And to "not reinvent the bicycle" I want to use existing ScriptAsAlter.  I believe your code should do the job.

Thanks a lot!
